packages/plugin-dom/DomRenderer.ts

```diff
@@ -122,54 +104,16 @@ export class DomRenderer extends Renderer<
             renderedDomNodes.push(placeholderBr);
         }
 
-        currentContext.parentNode.appendChild(fragment);
-
-        return new Map(
-            renderedDomNodes.map(domNode => {
-                return [domNode, [node]];
-            }),
-        );
-    }
-    /**
-     * Return the next rendering context, based on the given context. This
-     * includes the next VNode to render and the next parent element to render
-     * it into.
-     *
-     * @param currentContext
-     */
-    _nextRenderingContext(currentContext: DomRenderingContext): DomRenderingContext {
-        const node = currentContext.currentNode;
-        if (node.hasChildren()) {
-            currentContext.currentNode = node.firstChild();
-            // Render the first child with the current node as parent.
-            const renderedParent = VDocumentMap.toDom(node);
-            if (renderedParent) {
-                currentContext.parentNode = renderedParent;
-            }
-            this._contextStack.push({ ...currentContext });
-        } else if (node.nextSibling()) {
-            // Render the siblings of the current node with the same parent.
-            this._contextStack[this._contextStack.length - 1].currentNode = node.nextSibling();
-        } else {
-            // Render the next ancestor sibling in the ancestor tree.
-            let ancestor = node;
-            // Climb up the ancestor tree to the first parent having a sibling.
-            while (ancestor && !ancestor.nextSibling() && ancestor != currentContext.root) {
-                ancestor = ancestor.parent;
-                this._contextStack.pop();
+        // Render the node's children
+        let rendered = [];
+        node.children.forEach(child => {
+            if (!rendered.includes(child)) {
+                rendered = [...rendered, ...this.renderNode(child, renderedElement).keys()];
             }
-            if (ancestor && ancestor != currentContext.root) {
-                // At this point, the found ancestor has a sibling.
-                this._contextStack[
-                    this._contextStack.length - 1
-                ].currentNode = ancestor.nextSibling();
-            } else {
-                // If no next ancestor having a sibling could be found then the
-                // tree has been fully rendered.
-                return;
-            }
-        }
-        return this._contextStack[this._contextStack.length - 1];
+        });
+
+        domParent.appendChild(fragment);
```

Comment:
I think the very point of backing down to a "rendering a node renders its children" mechanism is to actually do the append in the parent itself. So the api of `render` would just be `render<T>(node: VNode): T` and then the parent would take the result and append it where he wants it to be appended. The point is the parent controls where the children is placed.